PocketKit/Sources/Analytics/AppEvents/Premium.swift

```diff
@@ -0,0 +1,97 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import SharedPocketKit
+
+public extension Events {
+    struct Premium {}
+}
+
+public extension Events.Premium {
+    /// Premium Upgrade View appears
+    /// - Parameter source: "settings" or "search"
+    static func premiumUpgradeViewShown(source: String) -> Event {
```

Comment:
Let's turn this into an enum. Let's make our analytics super safe and not freeform.

Reply:
> I'd like to request changes on the enum stuff I mentioned.
> 
> I also see a bunch of analytics added, can we add them to the UI tests or followup with that in another pr?

Thanks for reviewing @bassrock ! Enum done, and I will have a follow up pr for UI tests